---
title: 'Comparative Analysis of SARS-Cov-2 Haut-de-France Viral Sequenced Samples'
author: "Vivian Lynn Ly"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: pdf_document
csl: bioinformatics.csl
bibliography: references.bib
params:
  gff_file_path: "data/metadata/sars_refgenome_annotation.gff"
  vcf_dir_path: "data/11_vcf_output_for_R"
  sra_runtable_path: "data/00_sra_runtable/SraRunTable.txt"
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(echo = FALSE,
                      warning = FALSE,
                      message = FALSE,
                      error = TRUE)
```

# Background and Overview
This is a report on SARS-Cov-2, including some variant analysis [@koyama2020variant].

This rmarkdown file will use the 'vcfR'[@vcfR], ggplot2'[@ggplot2], 'dplyr'[@dplyr], and 'covdata'[@covdata] packages in order to analyze positive sequenced Sars-Cov-2 samples from the Haute-de-France after it has been processed: 

Haute-de-France directly translates to the North of France is comprised of 5 cities: Oise, Aisne, Nord, Pas-de-Calais, and Somme. Towards the beginning of the pandemic back in March 2020, France recorded about only about 30 confirmed cases of the Corona-virus. However, the few confirmed cases of the Corona-virus does not explain the huge spike in Covid-related deaths. More analysis is required to understand how cases and reported deaths have increased.

More specifically, this pandemic has tremendously affected the wine industry, suburban neighborhoods, holidays, rituals, and most importantly France's economy. The objective of this analysis is to thoroughly comprehend how variants of Sars-Cov-2 has affected hospitalizations due to COVID and recorded deaths due to COVID by analyzing mobility information and variant tracking of genes. 

How has mobility and gene variation in Haute-de-France positive sequenced samples affect the cases of hospitalization and deaths due to COVID?


# Methods

See the set of tutorials on the [vcfR package website](https://knausb.github.io/vcfR_documentation/index.html).

You may also want to use any of a range of different COVID data packages and data sources:

* https://kjhealy.github.io/covdata/
* https://github.com/como-ph/oxcovid19
* https://ropensci.org/blog/2020/10/20/searching-medrxivr-and-biorxiv-preprint-data/
* https://covidtracking.com/data/api
  * `readr::read_csv("https://api.covidtracking.com/v1/states/daily.csv")`
  
In order to compare how the Corona-virus has progressed within 2019-2021, I extracted information of cases and deaths related to Sars-Cov-2 variants from covdata's database. The 'vcfR' [@vcfR] package then will be used  visualize, manipulate, and filter quality in vfcR files after it has been processed through the makefile within the git repository. Afterwards, I will use 'ggplot2' [@ggplot2] and 'ggthemes'[@ggthemes] to create visual representations of the processed samples from the Haute-de-France. The 'dplyr' [@dplyr] package will be used to further filter information from the provided datasets.


# Results and Discussion

```{r load-packages-and-functions}
library("vcfR")
library("ggplot2")
library("dplyr")
library("ggthemes")
library("covid19.analytics")
library("covdata")

# this loads all function files in the code/functions directory
function_paths <- list.files("code/functions", full.names = TRUE)
purrr::walk(function_paths, source)
```

```{r load-vcf-data}
# load in, tidy, and stack all of the VCF files in the target directory
# these are the output of the bash pipeline
stacked_vcfs <- parse_tidy_and_stack_vcfs(
  vcf_dir_path = params$vcf_dir_path)

# load in the gff file with genome annotations for SARS-CoV-2
gff <- read_gff(gff_file_path = params$gff_file_path)

# pull out just the named genes from the annotation file
gene_table <- extract_genes_from_gff(annotation_object = gff)

# combine the stacked vcf table with the gene names and the metadata from
# the sra runtable file
vcf_with_metadata <- add_genes_metadata_to_vcfstack(
  sra_runtable_path = params$sra_runtable_path,
  stacked_vcf = stacked_vcfs,
  cleaned_genes_table = gene_table)
```

# Figures

```{r count-of-distinct-SNPs-in-Haute-de-France-Samples}
vcf_with_metadata %>%
# create a plot of unique SNP locations within each gene across all samples
  filter(!is.na(gene)) %>%
# get rid of SNPs not in gene regions
  group_by(gene, pos) %>%
  tally() %>%
# this gives a column n for the number of gene by position
  group_by(gene) %>%
  tally() %>%
# this collapses that down to the number of unique SNP locations
  ggplot(aes(x = gene,
             y = n)) +
    geom_col() +
    labs(title = "Count of distinct SNPs in Haute-de-France SARS-CoV-2 Genes",
         x = "Gene Name",
         y = "Number of Distinct SNPs") +
  theme_few() # get rid of the grey background
```

**Figure 1**: According to this figure which captures the unique SNPs locations within genes across all the sequenced samples, the N gene, S gene, and ORF3a gene contain the most SNPs location susceptible to mutation.  

```{r Transit-Mobility-Data-Nord-Pas-de-Calais Region, France}
# Using apple mobility data set to observe how mobility in the
# Nord-Pas-de-Calais Region in France compares to the recorded cases and
# deaths from Covdata.

france_mobility <- apple_mobility %>%
# Using apple mobility dataset to find France mobility
# data for transportation by Transit
  filter(region == "Nord-Pas-de-Calais Region") %>%
# This statement filters the information to the
# Northern Region of France where Haut-de-France is located.
  filter(transportation_type == "transit") %>%
# Filter transportation type to equal only transit
  filter(date >= as.Date("2020-03-01") & date <= as.Date("2021-05-01"))
# I decided to filter the date to only include mobility from
# when the epidemic began until now.

france_mobility %>%
  ggplot(aes(x =  france_mobility$date,
             y =  france_mobility$score)) +
#ggplot is being used to graph the mobility score based on dates.
  geom_line() +
  labs(title = "Northern Region of France Mobility during COVID(Transit)",
       x = "Date",
       y = "Transportation by Transit Levels")
# Creating a title for my diagram and labeling axis x and y.
```
**Figure 2**: In this figure that diagrams transportation by transit mobility from the apple mobility dataset, there are two significant peaks that occur from July 2020 - October 2020. Compared to the visualization that illustrates the cases and numbers of deaths, the mobility by transit transportation correlates with the the trend shown in Figure 8 and Figure 9. However, this does not explain how the the amount of covid cases towards the beginning of this epidemic does not correlate to the amount of deaths that occurred. Further analysis is performed to examine this characteristic. 

```{r Walking-Mobility-Data-Nord-Pas-de-Calais Region, France}
# Using apple mobility data set to observe how mobility in the
# Nord-Pas-de-Calais Region in France compares to the recorded
# cases and deaths from Covdata.

france_mobility <- apple_mobility %>%
# Using apple mobility dataset to find France mobility data for
# transportation by walking.
  filter(region == "Nord-Pas-de-Calais Region") %>%
# This statement filters the information to the Northern Region
# of France where Haut-de-France is located.
  filter(transportation_type == "walking") %>%
# Filter transportation type to equal only walking
  filter(date >= as.Date("2020-03-01") & date <= as.Date("2021-05-01"))
# I decided to filter the date to only include mobility from when
# the epidemic began until now.

france_mobility %>%
  ggplot(aes(x =  france_mobility$date,
             y =  france_mobility$score)) +
#ggplot is being used to graph the mobility score based on dates.
  geom_line() +
  labs(title = "Northern Region of France Mobility during COVID (Walking)",
       x = "Date",
       y = "Transportation by Walking Levels")
# Creating a title for my diagram and labeling axis x and y.
```
**Figure 3**: The figure illustrates transportation by walking mobility levels from the apple mobility data examining the Nord-Pas-de-Calais region of France. Similar to the previous figure, this diagram demonstrates a correlation between transportation by walking and the number of cases and deaths related to Covid-19. Unlike the previous figure, Figure 2, this visual shows us further indication of how there could have been a spike earlier on in the number of recorded deaths related to Covid-19. As we can observe from this figure, there are three prominent peaks that occur in March 2020, July 2020, and October 2020. However, this information still does not correlate with how low the numbers of confirmed Coronavirus cases there were in the beginning when the epidemic first began.

```{r Driving-Mobility-Data-Nord-Pas-de-Calais Region, France}
# Using covdata to observe driving mobility in France during the Covid Pandemic

france_mobility <- apple_mobility %>%
# Using apple mobility dataset to find France mobility data
# for transportation by walking.
  filter(region == "Nord-Pas-de-Calais Region") %>%
# This statement filters the information to the Northern
# Region of France where Haut-de-France is located.
  filter(transportation_type == "driving") %>%
# Filter transportation type to equal only driving
  filter(date >= as.Date("2020-03-01") & date <= as.Date("2021-05-01"))
# I decided to filter the date to only include mobility from
# when the epidemic began until now.

france_mobility %>%
  ggplot(aes(x =  france_mobility$date,
             y =  france_mobility$score)) +
#ggplot is being used to graph the mobility score based on dates.
  geom_line() +
  labs(title = "Northern Region of France Mobility during COVID (Driving)",
       x = "Date",
       y = "Transportation by Driving Levels")
# Creating a title for my diagram and labeling axis x and y.

```
**Figure 4**: This figure demonstrates tranportation by dirivng mobility levels over the course of 2020 until 2021 during the Corona-virus epidemic in the Northern Region of France. From the visual illustrations of this figure, we can come to the conclusion that driving is much more preferable than taking the transit or walking. Additionally, given the patterns of the information presented in this diagram, we can associate this information with the numbers of confirmed cases and deaths related to the novel Corona-virus. 

```{r Tracking-S-Gene-Variation-Over-Time}
# Tracking the variation of the S gene over the course of the pandemic
vcf_with_metadata %>%
# Create a diagram of SNPs recoreded in the vcf file that were
# derived from processing our data in the makefile.
  filter(!is.na(gene), gene == "S") %>%
# Getting rid of snps that are not located within a gene.
  group_by(Collection_Date, gene) %>%
# Group by the collection date and the gene.
  tally() %>%
# Counting variation of the S gene based on the collection date.
  ggplot(aes(x = (Collection_Date),
             y = n)) +
# Used to visualize the variation of the S gene over the
# different collection dates on this dot plot.
geom_point() +
    labs(title = "Tracking S Gene Variation Over Time",
         x = "Date",
         y = "Variation of the S Gene")
# Creating a title for my diagram and labeling axis x and y.

```
**Figure 5**: This figure demonstrates the amount of variation amongst the s-gene of the Sars-Cov-2 Haute-de-France positive sequenced samples based on the collection dates. As we can see over the period of the different collection dates the variation of S-gene increases. This indicates a susceptibility of possibly more mutations that can allow Sar-Cov-2 to become even more transmissible. 

```{r Tracking-N-Gene-Variation-Over-Time}
# Tracking the variation of the N gene over the course of the pandemic
vcf_with_metadata %>%
# create a diagram of SNPs recoreded in the vcf file that
# were derived from processing our data in the makefile.
  filter(!is.na(gene), gene == "N") %>%
# Getting rid of snps that are not located within a gene.
  group_by(Collection_Date, gene) %>%
# Group by the collection date and the gene.
  tally() %>%
# Counting variation of the N gene based on the collection date.
  ggplot(aes(x = (Collection_Date),
             y = n)) +
# Used to visualize the variation of the N gene over the
# different collection dates on this dot plot.
geom_point() +
    labs(title = "Tracking N Gene Variation Over Time",
         x = "Date",
         y = "Variation of the N Gene")
# Creating a title for my diagram and labeling axis x and y.
```
**Figure 6**: Looking at the variation of the N-gene of Sars-Cov-2 Haute-de-France positive sequenced samples from the beginning of the pandemic until now. 

```{r Tracking-ORF3a-Gene-Variation-Over-Time}
# Tracking the variation of the ORF3a gene over the course of the pandemic
vcf_with_metadata %>%
# Create a diagram of SNPs recoreded in the vcf file that
# were derived from processing our data in the makefile.
  filter(!is.na(gene), gene == "ORF3a") %>%
# Getting rid of snps that are not located within a gene.
  group_by(Collection_Date, gene) %>%
# Group by the collection date and the gene.
  tally() %>%
# Counting variation of the ORF3a gene based on the collection date.
  ggplot(aes(x = (Collection_Date),
             y = n)) +
# Used to visualize the variation of the ORF3a gene over the
# different collection dates on this dot plot.
geom_point() +
    labs(title = "Tracking ORF3a Gene Variation Over Time",
         x = "Date",
         y = "Variation of the ORF3a Gene")
# Creating a title for my diagram and labeling axis x and y.
```
**Figure 7**: Looking at the variation of the ORF3a gene of Sars-Cov-2 Haute-de-France positive sequenced samples from the beginning of the pandemic until now. 

```{r France-data-weekly}
covnat_weekly %>%
  filter(cname == "France") %>%
ggplot(aes(x = date,
           y = cases)) +
  geom_line() +
  labs(title = "Amount of COVID cases Weekly in France",
       x = "Date",
       y = "Number of Positive Cases")

```
**Figure 8**: Tracking the number of COVID cases over time. 

```{r Tracking Deaths due to Covid in France}
covnat_weekly %>%
  filter(cname == "France") %>%
ggplot(aes(x = date,
           y = deaths)) +
  geom_line() +
  labs(title = "Tracking Amount of Deaths in France",
       x = "Date",
       y = "Number of Deaths")
```
**Figure 9**: Tracking the number of deaths related to COVID in France.

# Tables

```{r example-table}
# An example table to show the length of each gene using its start and end
gene_table %>%
  mutate(length = end - start) %>%
  select(gene_name, start, end, length) %>%
  knitr::kable(col.names = c("Gene Name",
                             "Start",
                             "End",
                             "Length"))
```

**Table 1**: Gene names, locations, and lengths in the SARS-CoV-2 genome. Higher SNP counts in the S and N genes may be related to the larger size of these genes.

```{r Alternate nucleotides within the S Gene}
# Top ten counts of alternate nucleotides compared to the
# reference at the S Gene
vcf_with_metadata %>%
  filter(!is.na(gene)) %>%
  group_by(gene, ref, alt) %>%
  tally() %>%
  select(gene, ref, alt, n) %>%
  filter(gene == "S") %>%
  arrange(desc(n)) %>%
  head(10) %>%
  knitr::kable(col.names = c("Gene",
                             "Reference",
                             "Alternate",
                             "Count"))
```
**Table 2**: Top Occurrence of alternate nucleotides compared to the reference at the S gene. 

```{r Alternate nucleotides within the N Gene}
# Top ten counts of alternate nucleotides compared to the
# reference at the S Gene
vcf_with_metadata %>%
  filter(!is.na(gene)) %>%
  group_by(gene, ref, alt) %>%
  tally() %>%
  select(gene, ref, alt, n) %>%
  filter(gene == "N") %>%
  arrange(desc(n)) %>%
  head(10) %>%
  knitr::kable(col.names = c("Gene",
                             "Reference",
                             "Alternate",
                             "Count"))

```
**Table 3**: Top Occurrence of alternate nucleotides compared to the reference at the N gene.

```{r Alternate nucleotides within the ORF3a Gene}
# Top ten counts of alternate nucleotides compared to
# the reference at the S Gene

vcf_with_metadata %>%
  filter(!is.na(gene)) %>%
  group_by(gene, ref, alt) %>%
  tally() %>%
  select(gene, ref, alt, n) %>%
  filter(gene == "ORF3a") %>%
  arrange(desc(n)) %>%
  head(10) %>%
  knitr::kable(col.names = c("Gene",
                             "Reference",
                             "Alternate",
                             "Count"))
```
**Table 5**: Top Occurrence of alternate nucleotides compared to the reference at the ORF3a gene.

# Sources Cited
